smt/utils/design_space.py

```diff
@@ -738,17 +744,23 @@ def _correct_get_acting(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
 
         return x_corr, is_acting
 
-    def _sample_valid_x(self, n: int) -> Tuple[np.ndarray, np.ndarray]:
+    def _sample_valid_x(
+        self, n: int, random_state=None, **kwargs
+    ) -> Tuple[np.ndarray, np.ndarray]:
         """Sample design vectors"""
 
         # Simplified implementation: sample design vectors in unfolded space
         x_limits_unfolded = self.get_unfolded_num_bounds()
-        sampler = LHS(xlimits=x_limits_unfolded, random_state=self.seed)
-        x = sampler(n)
+        self.seed = random_state
```

Comment:
Why creating `self.seed`? Why not just use `random_state`? One call to `sample_valid_x` with `random_state`, will alter behaviour of further calls without `random_state` which seems error-prone to me.